src/test.rs

```diff
@@ -384,3 +410,17 @@ fn test_so_linger() {
     };
     assert_eq!(val, 42);
 }
+
+#[test]
+fn test_reentrant() {
+    let null = HasWeirdSerializer(None);
+    let hello = HasWeirdSerializer(Some(String::from("hello")));
+    let (sender, receiver) = ipc::channel().unwrap();
+    let sender1 = sender.clone();
+    WEIRD_CHANNEL.with(|chan| { *chan.borrow_mut() = Some(sender1); });
```

Comment:
Nitpick: since this is not a move closure, it seem circuitous to do the `clone()` beforehand -- I think it would be cleaner simply to do `Some(sender.clone())` within the closure...

This is mostly just a general hint though -- as we are talking about test code here, I wouldn't insist on fixing such a minor stylistic wrinkle :-)